src/components/FxParams/Controls.tsx

```diff
@@ -64,36 +68,91 @@ export const Controls = ({
   onChangeData,
 }: ControlsProps) => {
   const consolidatedParams = consolidateParams(params, data)
-
   const p: React.RefObject<HTMLDivElement> = createRef()
+  const [collapsedGroups, setCollapsedGroups] = useState<{
+    [key: string]: boolean
+  }>({})
+
+  const toggleGroup = (groupName: string) => {
+    setCollapsedGroups((prev) => ({
+      ...prev,
+      [groupName]: !prev[groupName],
+    }))
+  }
 
   useEffect(() => {
     const ps: any = {}
-    if (consolidatedParams?.length > 0) {
```

Comment:
Not sure why this check was here but to reduce the area of potential failure of this PR we'd want to keep it, or explain how the check can be bypassed safely